core/src/main/java/org/elasticsearch/common/settings/Settings.java

```diff
@@ -1210,10 +1211,23 @@ public boolean shouldRemoveMissingPlaceholder(String placeholderName) {
             Iterator<Map.Entry<String, Object>> entryItr = map.entrySet().iterator();
             while (entryItr.hasNext()) {
                 Map.Entry<String, Object> entry = entryItr.next();
-                if (entry.getValue() == null || entry.getValue() instanceof List) {
+                if (entry.getValue() == null) {
                     // a null value obviously can't be replaced
                     continue;
                 }
+                if (entry.getValue() instanceof List) {
+                    List<String> ls = (List<String>) entry.getValue();
+                    ListIterator<String> li = ls.listIterator();
```

Comment:
Can this be a local final variable?

Reply:
+1 for `final`.  I am very clear what you meant by **local** ?  Do you recommend to declare `li` before the while loop on the line 1213?

Reply:
Sorry for not being clear. What I mean is that these variables (which are local variables, they only have scope of the current) block I think would clearer if they were final because they I know by the declaration they are never mutated instead of having to keep checking the code "is this the same object as when the variable was first declared?".